### PR TITLE
V4 docstrings

### PR DESCRIPTION
**Pull request by OrtnerMichael**
# Related Issues
#419 
#412 first steps only
#406 done
#403 code tasks

# Notes
- new input checks with properly formatted error messages
- `checkinputs` parameter is completely out, input checks are always performed
- not more tool tip codes `dimension=(a,b,c)` but only `dimension=None` inits
- complete docstings of V4 codes
- some minor code refactorings and mini-bugfixes

### CODE REVIEW

**Comment by Alexboiboi** [comment]
squash or merge. I'd squash

File: magpylib/_src/utility.py [hunk 1]
```diff
@@ -183,8 +182,12 @@ def format_obs_inputs(observers) -> list:
 
         # case 2: ndarray of positions
         elif isinstance(obs, (list, tuple, np.ndarray)):
-            if Config.checkinputs:
-                check_position_format(np.array(obs), "observer position")
+            check_format_input_vector(
+                obs,
+                dims=(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20),
```
Comment (Alexboiboi) [inline magpylib/_src/utility.py:187]: What is this???

**Review by Alexboiboi — CHANGES_REQUESTED** [review]

File: magpylib/_src/display/display.py [hunk 2]
```diff
@@ -96,17 +104,30 @@ def show(
 
     # test if all source dimensions and excitations have been initialized
     check_dimensions(obj_list_flat)
+    check_excitations(obj_list_flat)
 
     # test if every individual obj_path is good
     test_path_format(obj_list_flat)
 
+    # input checks
+    backend = check_format_input_backend(backend)
```
[the remark this answers is not in the document]
Comment (OrtnerMichael) [reply magpylib/_src/display/display.py:113]: I think we should have completely separate input checks where good and bad inputs are defined. I already made this an issue #406 .

Also I did not see an else statement in the `show` function.

[the remark this answers is not in the document]
Comment (OrtnerMichael) [reply magpylib/_src/display/display.py:113]: ah ... this did not exist when I forked off.

[the remark this answers is not in the document]
Comment (OrtnerMichael) [reply magpylib/_src/display/display.py:113]: fixed as proposed with new commit

File: magpylib/_src/display/display.py [hunk 3]
```diff
@@ -96,17 +104,30 @@ def show(
...
     if backend is None:
         backend = Config.display.backend
 
+    check_input_zoom(zoom)
+    check_input_animation(animation)
```
[the remark this answers is not in the document]
Comment (OrtnerMichael) [reply magpylib/_src/display/display.py:118]: fixed as above

File: magpylib/_src/input_checks.py [hunk 4]
```diff
@@ -4,238 +4,415 @@
...
-        raise MagpylibBadUserInput(msg)
-
-
-def check_start_type(start):
+def check_start_type(inp):
     """start input must be int or str"""
-    if not (isinstance(start, (int, np.int_)) or start == 'auto'):
-        msg = 'start input must be int or str ("auto")'
-        raise MagpylibBadUserInput(msg)
+    if not (isinstance(inp, (int, np.int_)) or inp == 'auto'):
```
[the remark this answers is not in the document]
Comment (OrtnerMichael) [reply magpylib/_src/input_checks.py:92]: fixed

[the remark this answers is not in the document]
Comment (OrtnerMichael) [reply magpylib/_src/input_checks.py:92]: done

File: magpylib/_src/input_checks.py [hunk 5]
```diff
@@ -4,238 +4,415 @@
...
-    if not (isinstance(anch, (list, tuple, np.ndarray, type(None))) or anch == 0):
-        msg = "anchor input must be 0, None or a vector (list, tuple or ndarray)."
-        raise MagpylibBadUserInput(msg)
+def check_input_animation(inp):
+    """check show animation input"""
+    ERR_MSG = (
+        "Input parameter `animation` must be boolean or a positive number.\n"
+        f"Instead received {inp}."
+        )
+    if not isinstance(inp, (bool, float, int, np.float_, np.int_)):
```
[the remark this answers is not in the document]
Comment (OrtnerMichael) [reply magpylib/_src/input_checks.py:79]: fixed

**Review by Alexboiboi — APPROVED** [review]

**Pull request merged**